packages/api-core/src/site-users/fetchers.ts

```diff
@@ -14,3 +14,17 @@ export async function fetchSiteUsers( siteId: number ): Promise< SiteUser[] > {
 		apiNamespace: 'wp/v2',
 	} );
 }
+
+export async function fetchWpcomSiteUsers(
+	siteId: number,
+	role?: string
```

Comment:
Nit: Since role is optional here, would it be better to make it as something like `options` object with defined properties?

```js
options?: { role?: string }

...
query: {
  force: 'wpcom',
  ...options
}
```